expansion/src/main/java/no/unit/nva/expansion/model/IndexDocumentWrapperLinkedData.java

```diff
@@ -13,41 +15,67 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.stream.Collectors;
+import no.unit.nva.commons.json.JsonUtils;
 import no.unit.nva.expansion.utils.FramedJsonGenerator;
 import no.unit.nva.expansion.utils.SearchIndexFrame;
 import no.unit.nva.expansion.utils.UriRetriever;
 import nva.commons.core.ioutils.IoUtils;
 
 public class IndexDocumentWrapperLinkedData {
-    
+
+    public static final String PART_OF_FIELD = "/partOf";
+    public static final String ID_FIELD = "/id";
     private final UriRetriever uriRetriever;
-    
+
+    private static final String ORGANIZATION_TYPE = "Organization";
+    private static final String TYPE_FIELD = "type";
+
+
     public IndexDocumentWrapperLinkedData(UriRetriever uriRetriever) {
         this.uriRetriever = uriRetriever;
     }
-    
+
     public String toFramedJsonLd(JsonNode indexDocument) throws IOException {
         String frame = SearchIndexFrame.FRAME_SRC;
         List<InputStream> inputStreams = getInputStreams(indexDocument);
         return new FramedJsonGenerator(inputStreams, stringToStream(frame)).getFramedJson();
     }
-    
+
     private List<InputStream> getInputStreams(JsonNode indexDocument) {
         final List<InputStream> inputStreams = new ArrayList<>();
         inputStreams.add(stringToStream(toJsonString(indexDocument)));
         inputStreams.addAll(fetchAll(extractPublicationContextUris(indexDocument)));
+        inputStreams.addAll(fetchAll(extractAffiliationUris(indexDocument)));
         return inputStreams;
     }
-    
+
     private Collection<? extends InputStream> fetchAll(List<URI> publicationContextUris) {
-        List<Optional<String>> uriContent =
-            publicationContextUris.stream().map(this::fetch).collect(Collectors.toList());
+        List<Optional<String>> uriContent = new ArrayList<>();
+
+        for (URI uri : publicationContextUris) {
+            var content = fetch(uri);
+            uriContent.add(content);
+
+            content.ifPresent(cont -> fetchChildUrisForOrganizations(uriContent, cont));
+        }
+
         return uriContent.stream()
-                   .flatMap(Optional::stream)
-                   .map(IoUtils::stringToStream)
-                   .collect(Collectors.toList());
+            .flatMap(Optional::stream)
+            .map(IoUtils::stringToStream)
+            .collect(Collectors.toList());
     }
-    
+
+    private void fetchChildUrisForOrganizations(List<Optional<String>> uriContent, String content) {
```

Comment:
Why not let the function take in List<String>. It should be the callers responsibility to remove non-valid fields